### PR TITLE
Update dispatchEventWithResponseCallback API with timeout param

### PR DESCRIPTION
**Pull request by cacheung**
Add dispatchEventWithResponseCallback(event: Event, timeoutMS: Number): Promise<Event> ()

Remove dispatchEventWithResponseCallback(event: Event): Promise<Event> ()

Including few changes from the earlier PR review.
Add dispatchEvent API back

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/core/__tests__/MobileCoreTests.ts [hunk 1]
```diff
@@ -76,7 +85,7 @@ describe('MobileCore', () => {
     let testEvent = new Event('eventName', 'eventType', 'eventSource', {
       testDataKey: 'testDataValue'
     });
-    await MobileCore.dispatchEventWithResponseCallback(testEvent);
+    await MobileCore.dispatchEventWithResponseCallback(testEvent, 5000);
     expect(spy).toHaveBeenCalledWith(testEvent);
```
Comment (yangyansong-adbe) [inline packages/core/__tests__/MobileCoreTests.ts:89]: `dispatchEventWithResponseCallback` is called with two arguments. Let's add the timeout argument as well.

File: packages/core/ts/MobileCore.ts [hunk 2]
```diff
@@ -145,17 +146,29 @@ const MobileCore: IMobileCore = {
     return RCTAEPCore.getSdkIdentities();
   },
 
+   /**
+   * Called by the extension public API to dispatch an event for other extensions or the internal SDK to consume.
+   * Any events dispatched by this call will not be processed until after `start` has been called.
+   *
+   * @param event Required parameter with {@link Event} instance to be dispatched. Should not be nil
```
Comment (yangyansong-adbe) [inline packages/core/ts/MobileCore.ts:153]: missing `.`

File: packages/core/ts/MobileCore.ts [hunk 3]
```diff
@@ -145,17 +146,29 @@ const MobileCore: IMobileCore = {
...
+   * @return true if the the event dispatching operation succeeded, otherwise the promise will return an error
```
Comment (yangyansong-adbe) [inline packages/core/ts/MobileCore.ts:154]: missing `.`

File: packages/core/ts/MobileCore.ts [hunk 4]
```diff
@@ -145,17 +146,29 @@ const MobileCore: IMobileCore = {
...
+   */
+   dispatchEvent(event: Event): Promise<boolean> {
+    return RCTAEPCore.dispatchEvent(event);
+  },
```
Comment (yangyansong-adbe) [inline packages/core/ts/MobileCore.ts:158]: nit: The code format seems to be incorrect. Let's adjust it manually.

File: packages/core/ts/MobileCore.ts (hunk 3 above)
Comment (cacheung) [reply packages/core/ts/MobileCore.ts:154]: I added the periods.  But in this file not param and return have periods after the sentences.

**Review by yangyansong-adbe — APPROVED** [review]

**Pull request merged**